libs/community/langchain_community/vectorstores/lancedb.py

```diff
@@ -52,25 +57,41 @@ def __init__(
                 "Could not import lancedb python package. "
                 "Please install it with `pip install lancedb`."
             )
-        self.lancedb = lancedb
         self._embedding = embedding
         self._vector_key = vector_key
         self._id_key = id_key
         self._text_key = text_key
         self._table_name = table_name
+        self.api_key = api_key
+        self.region = region
+        self.mode = mode
+
+        if os.getenv("LANCE_API_KEY") is not None:
+            api_key = os.getenv("LANCE_API_KEY")
+
+        if "db://" in uri and api_key is None:
+            raise ValueError("API key is required for LanceDB cloud.")
 
         if self._embedding is None:
-            raise ValueError("embedding should be provided")
+            raise ValueError("embedding object should be provided")
 
-        if connection is not None:
-            if not isinstance(connection, lancedb.db.LanceTable):
-                raise ValueError(
-                    "connection should be an instance of lancedb.db.LanceTable, ",
-                    f"got {type(connection)}",
-                )
+        if isinstance(connection, lancedb.db.LanceDBConnection):
```

Comment:
This will break previous version rigght? Previous versions had connections objects as lanceDBTable object right?

Reply:
yes , the previous version was not designed well , this is a better design.

Reply:
No but how will users know. Let's add a deprication warning first if they pass a table object saying it won't work as expected.

Reply:
cool will add a warn